CONTRIBUTING.md

```diff
@@ -0,0 +1,324 @@
+# Contributing Guide
+
+Thank you for considering contributing to SG-NLP! We believe that SG-NLP's future is closely tied with the community and community contributions will help SG-NLP to grow faster.
+
+## How can I contribute?
+
+- [Add a new model to the `sgnlp` package](#adding-a-new-model-to-sgnlp)
+- Add a new demo for an existing model to the SG-NLP demo website - refer to the contributing guide for the NLP Hub demo.
+- [Submit bug fixes](#submitting-bug-fixes)
+- [Add documentation](#adding-documentation)
+
+## Adding a new model to `sgnlp`
+
+### Pre-requisites
+
+1. Fork the `sgnlp` repository.
+2. Create a Python virtual environment (verison >= 3.8)
+3. Install the packages in `requirements_dev.txt` at the root of the repository using `pip install -r requirements_dev.txt`. This will install `black`, `flake8`, and `pytest` which we use for code formatting and testing.
+
+### Required Components
+
+Before you create a pull request to add your model to `sgnlp`, please ensure that you have the following components ready.
+
+- Python scripts / code comprising of (we'll go into more detail below)
+  - [config.py](#config)
+  - [modeling.py](#modeling)
+  - [preprocess.py](#preprocess)
+  - [train.py](#train)
+  - [eval.py](#eval)
+  - [utils.py](#utils) (optional)
+  - [README](#readme)
+  - [requirements.txt](#requirements.txt) (discouraged)
+- Model information (to be included in the README)
+  - Original paper / source
+  - Datasets and/or how to obtain them
+  - Evaluation metrics
+  - Model size
+  - Training information
+- Model weights and artefacts (to be submitted separately)
+  - pytorch_model.bin
+  - config.json
+  - tokenizer_config.json (optional)
+
+### Project Structure
+
+The code for your model should follow this structure.
+
+`sgnlp`  
+`---- sgnlp`  
+`-------- models`  
+`------------ model_name`  
+`---------------- config`  
+`---------------- modules (optional)`  
+`---------------- README.md`  
+`---------------- __init__.py`  
+`---------------- config.py`  
+`---------------- eval.py`  
+`---------------- modeling.py`  
+`---------------- tokenization.py (optional)`  
+`---------------- train.py`  
+`---------------- utils.py (optional)`  
+`---------------- requirements.txt (optional)`
```

Comment:
Project structure is a little weird. Maybe go with something like

sgnlp
|-- sgnlp
|--  |-- models

On another note, they should be forking and shouldn't need to create the entire folder structure from scratch.
So maybe can be simplified to say create a folder in sgnlp/models/<model_name> with the following files.

Overall, will prefer simplicity.